compendium/modules/w06-patterns-exercise.tex

```diff
@@ -281,6 +302,49 @@
 \QUESTEND
 
 
+\WHAT{Mönstermatcha med nyttan av \code{enum}.}
+
+\QUESTBEGIN
+\TODO %Se gärna över denna frågan samt facit.
+\Task	\what~ Vi ska nu undersöka och jämföra skillnad mellan nyckelorden \code{enum} och \code{sealed}. Skriv nedan kod i en REPL.
+\begin{Code}
+enum Färg:
+  case Spader, Hjärter, Ruter, Klöver
+\end{Code}
+
+\Subtask Skapa med hjälp av en editor igen en funktion \code{def paraFärg(f: Färg): Färg}, nästintill likadan som den som vi skapade i deluppgift \ref{task:match-sealedtrait-caseobject} \ref{subtask:match-sealedtrait}. Funktionen ska återigen utnyttja match-uttryck för att returnera paralellfärgen till argumentet som ges. Tänk på att denna gången är \code{Färg} inget \code{sealed trait}, utan istället en \code{enum}. Klistra in funktionen i REPL.
+\begin{REPL}
+scala> parafärg(Färg.Ruter)
+scala> val xs = Vector.fill(5)(Färg.values((math.random() * 4).toInt))
+scala> xs.map(parafärg)
+\end{REPL}
+
+
+\Subtask
+Fundera på skillnader och likheter mellan att utnyttja \code{sealed trait} ihop med \code{case}-objekt gentemot att använda sig av \code{enum} vid mönstermatchning.
+
+
+\SOLUTION
+
+
+\TaskSolved \what
+\TODO %Se över denna
+\SubtaskSolved
+\begin{Code}
+def parafärg(f: Färg): Färg = f match
+  case Färg.Spader  => Färg.Klöver
+  case Färg.Hjärter => Färg.Ruter
+  case Färg.Ruter   => Färg.Hjärter
+  case Färg.Klöver  => Färg.Spader
+\end{Code}
+
```

Comment:
Add a comment that you can use import to skip `Färg.` but it is often better to keep it to avoid name clashes. Compare if you had several color Red one for cards and one for Pixels...